File: LeetCode/Easy/3010-divide-an-array-into-subarrays-with-minimum-cost-i/3010-divide-an-array-into-subarrays-with-minimum-cost-i.cpp
class Solution {
public:
    int minimumCost(vector<int>& nums) {
        int ret = nums[0];
        int first = 999;
        int second = 999;
        
        for (int i = 1; i < nums.size(); ++i){
            int temp = nums[i];
            if (temp < first){
                second = first;
                first = temp;
            } else if (temp < second) second = temp;
        }
        
        return ret + first + second;
    }
};

File: LeetCode/Easy/3010-divide-an-array-into-subarrays-with-minimum-cost-i/3010-divide-an-array-into-subarrays-with-minimum-cost-i.go
func minimumCost(nums []int) int {
    ret := nums[0]
    left := 2
    count := make([]int, 51)

    for i := 1; i < len(nums); i++ { count[nums[i]]++ }

    i := 1

    for left > 0 {
        if count[i] > 0 {
            count[i]--
            ret += i
            left--
        } else { i++ }
    }

    return ret
}